app/src/marketplace/modules/contractProduct/services.js

```diff
@@ -125,3 +133,41 @@ export const getSubscribedEvents = async (id: ProductId, fromTimestamp: number,
 
     return subscriptions
 }
+
+const createOrUpdateContractProduct = (method: (...any) => Sendable, product: SmartContractProduct): SmartContractTransaction => {
+    const {
+        id,
+        name,
+        beneficiaryAddress,
+        pricePerSecond,
+        priceCurrency,
+        minimumSubscriptionInSeconds,
+    } = product
+    const currencyIndex = Object.keys(currencies).indexOf(priceCurrency)
+    validateContractProductPricePerSecond(pricePerSecond)
+    validateProductPriceCurrency(priceCurrency)
+    const transformedPricePerSecond = mapPriceToContract(pricePerSecond)
+    const methodToSend =
+        method(getValidId(id), name, beneficiaryAddress, transformedPricePerSecond, currencyIndex, minimumSubscriptionInSeconds, false)
```

Comment:
I will put out a separate fix for this, extra `boolean` parameter is needed to make the call.